src/ngraph/frontend/onnx_import/op/quant_conv.cpp

```diff
@@ -0,0 +1,248 @@
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#include "ngraph/builder/quantization/quantized_linear_convolution.hpp"
+#include "ngraph/frontend/onnx_import/exceptions.hpp"
+#include "ngraph/frontend/onnx_import/op/conv.hpp"
+#include "ngraph/frontend/onnx_import/utils/broadcasting.hpp"
+#include "ngraph/frontend/onnx_import/utils/convpool.hpp"
+#include "ngraph/op/add.hpp"
+#include "ngraph/op/broadcast.hpp"
+#include "ngraph/op/concat.hpp"
+#include "ngraph/op/divide.hpp"
+#include "ngraph/op/experimental/quantized_conv.hpp"
+#include "ngraph/op/multiply.hpp"
+#include "ngraph/op/slice.hpp"
+#include "quant_conv.hpp"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        namespace op
+        {
+            namespace set_1
+            {
+                namespace
+                {
+                    struct OpScale
+                    {
+                        std::shared_ptr<ngraph::Node> data_scale;
+                        std::shared_ptr<ngraph::Node> filter_scale;
+                        std::shared_ptr<ngraph::Node> output_scale;
+                    };
+
+                    struct OpZeroPoint
+                    {
+                        std::shared_ptr<ngraph::Node> data_zero_point;
+                        std::shared_ptr<ngraph::Node> filter_zero_point;
+                        std::shared_ptr<ngraph::Node> output_zero_point;
+                    };
+
+                    std::shared_ptr<ngraph::Node>
+                        make_ng_quant_conv(const std::shared_ptr<ngraph::Node>& data,
+                                           const std::shared_ptr<ngraph::Node>& filters,
+                                           const ngraph::Strides& strides,
+                                           const ngraph::Strides& filter_dilations,
+                                           const ngraph::CoordinateDiff& padding_below,
+                                           const ngraph::CoordinateDiff& padding_above,
+                                           const ngraph::Strides& data_dilations,
+                                           int groups,
+                                           const OpScale& op_scale,
+                                           const OpZeroPoint& op_zero_point)
+                    {
+                        if (groups > 1)
+                        {
+                            // Split one convolution op to N ops where N is the number of groups
+                            // and concat results after computation.
+                            // reference: https://github.com/NervanaSystems/ngraph-mxnet/blob/fdd692/src/ngraph/ngraph_emitter.cc#L822-L856
+                            std::size_t n_data_channels{data->get_shape().at(1)};
+                            std::size_t n_filters_channels{filters->get_shape().at(0)};
+                            // TODO: ensure n_data_channels % groups = 0
```

Comment:
Add check from this comment. This should have been already added in conventional convolutions.